docs/graph_explorer.html

```diff
@@ -90,11 +91,11 @@
           <div id="editorBox">
             <pre id="editor"></pre>
           </div>
-          <div id="editorHelp">
+          <article id="editorHelp">
```

Comment:
this isn't an article

Reply:
> most of this seems fine but the `<article>` and `<section>` things are semantics for the sake of semantics, and these don't fit those semantic usages

Actually, they do. :)


## `article`

Here’s an excerpt from the [HTML5 Spec on `article`](https://html.spec.whatwg.org/multipage/sections.html#the-article-element):

> The `article` element represents a complete, or self-contained, composition in a document, page, application, or site and that is, in principle, independently distributable or reusable, e.g. in syndication. This could be a forum post, a magazine or newspaper article, a blog entry, a user-submitted comment, an interactive widget or gadget, or any other independent item of content.
>
> When `article` elements are nested, the inner `article` elements represent articles that are in principle related to the contents of the outer `article`. For instance, a blog entry on a site that accepts user-submitted comments could represent the comments as `article` elements nested within the `article `element for the blog entry.

The spec contains examples, too.  

The [MDN page on `article`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article) has a (verbatim from the spec?) introduction, along with [its own examples](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article#examples).


## `section`

From the [HTML5 spec on `section`](https://html.spec.whatwg.org/multipage/sections.html#the-section-element):

> <b>Note</b> The `section` element is not a generic container element. When an element is needed only for styling purposes or as a convenience for scripting, authors are encouraged to use the `div` element instead. A general rule is that the `section` element is appropriate only if the element's contents <em>would be listed explicitly in the document's outline.</em> [emphasis added by me]

Since the help document uses collapsible headings, that’s pretty much an interactive outline as far as I can tell.

----

Regarding “semantics for the sake of semantics”: I absolutely agree that there is such a thing as going too far.  However, these particular semantic changes are pretty minimal, and an appropriate use for the HTML5 elements.  

A good chunk of HTML5 was designed by looking at the `id` and `class` names given to all the `div` elements on the web.  Articles and sections were among the most common, so that’s why they’re in the spec.  In a sense, they’re purpose-specific refinements of the `div` element.

----

If you have another reason for disliking these elements, that’s fine.  Or, if you dislike them for no reason at all—it’s your project!  It’s not my place to tell you what to like or dislike.  

_But_, they are semantically appropriate.  :)  

I care deeply for semantic HTML, and I would never knowingly abuse the semantics of an element.  If I felt the appropriate use of a particular element was in something of a gray area, I would try to point that out.  There’s no shortage of subjectivity—not even in programming. (Ha… _Especially_ not in programming!)

Reply:
this isn't a self contained composition

this is not a document

there is no outline

your quote explicitly says don't do this

> If you have another reason for disliking these elements, that’s fine.

i don't want to argue.  these are semantically incorrect

Reply:
this is like when people used to insist that every menu needed to be an unordered list and then pretend that it was about semantics, while bending over wildly backwards to stretch semantic definitions to fit

this is an application.  article is for articles - text documents.  article isn't even appropriate for most text documents; they aren't articles either.

Reply:
Ouch.  

Okay, fine. It’s your project.  I’ll revert the changes.